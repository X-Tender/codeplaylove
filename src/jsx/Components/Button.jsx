import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Button = ({ children, isPrimary, isGhost, isFullWithSm, to }) => {
	let style = '';

	if (isPrimary) style += 'button--primary ';
	if (isGhost) style += 'button--ghost ';
	if (isFullWithSm) style += 'button--full-width-sm ';

	return (
		<Link className={`button ${style}`} to={to}>
			{children}
		</Link>
	);
};

Button.propTypes = {
	to: PropTypes.string.isRequired,
	children: PropTypes.any,
	isPrimary: PropTypes.bool,
	isGhost: PropTypes.bool,
	isFullWithSm: PropTypes.bool,
};

Button.defaultProps = {
	children: null,
	isPrimary: false,
	isGhost: false,
	isFullWithSm: false,
};

export default Button;
