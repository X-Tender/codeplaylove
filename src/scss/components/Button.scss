.button {
	position: relative;
	display: inline-block;
	padding: 15px 25px;
	border-radius: 50px;
	text-align: center;
	transition: background-color $defaultTransitionDuration ease, color $defaultTransitionDuration ease;

	&--primary {
		background-color: $primaryColor;
		border-color: $primaryColor;
		color: $mainBackgroundColor;

		&:after {
			content: '';
			display: block;
			position: absolute;
			size: 100%;
			border-radius: 50px;
			left: 0;
			top: 0;
			z-index: -1;
			box-shadow: 0 5px 30px $primaryColor;
			opacity: 0;
			transition: opacity $defaultTransitionDuration ease;
			will-change: opacity;
		}

		&:hover:after {
			opacity: 1;
		}
	}

	&--ghost {
		background-color: transparent;
		border-width: 2px;
		border-style: solid;
		color: $mainTextColor;

		&:hover {
			background-color: $primaryColor;
			color: $mainBackgroundColor;
		}
	}

	&--full-width-sm {
		display: block;
		@include sm {
			display: inline-block;
		}
	}
}
