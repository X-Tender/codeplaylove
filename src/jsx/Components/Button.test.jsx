import React from 'React';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
	const to = '/target';
	const children = 'The target';
	const isPrimary = true;
	const isGhost = true;
	const isFullWithSm = true;

	const buttonMinProps = shallow(<Button to={to} />);
	const buttonAllProps = shallow(
		<Button isFullWithSm={isFullWithSm} isGhost={isGhost} isPrimary={isPrimary} to={to}>
			{children}
		</Button>
	);

	it('renders correctly', () => {
		expect(buttonMinProps).toMatchSnapshot();
	});

	it('renders the correct props', () => {
		expect(
			buttonAllProps.find('.button.button--primary.button--ghost.button--full-width-sm').exists()
		).toBe(true);
	});
});
