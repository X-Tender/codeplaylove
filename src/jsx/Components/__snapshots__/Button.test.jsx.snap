// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Link
  className="button "
  replace={false}
  to="/target"
/>
`;
